### ISSUE TITLE
Calling`getClass` with SmallestModules module split style causes reference error

### ISSUE THREAD

**Issue by joprice** [scala-js/scala-js#4616]
When modules split style is set to `SmallestModules` and the code contains a reference to `getClass`, a ReferenceError is thrown:

```
target/scala-2.13/app-fastopt/java.lang.Object.js:819
    this._classOf = new $j_java$002elang$002eClass.$c_jl_Class(this)
                        ^

ReferenceError: $j_java$002elang$002eClass is not defined
```

Here is a repo that reproduces it:  https://github.com/joprice/scalajs-test. Running `sbt run` is enough to cause the error.

## versions
sbt: 1.5.3
sbt-scalajs: 1.8.0
scala: 2.13.7

**Comment by sjrd** [comment]
Thanks for the report and for the minimized reproduction. I am currently on vacation, but I will look at it when I'm back.

**Comment by gzm0** [comment]
I think the problem is that this line is not accounted for in dependency analysis:

https://github.com/scala-js/scala-js/blob/b0f1dc501edb15203176c636d35323c2bcc24550/linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala#L1669

Basically: If `java.lang.Class` is instantiated, Scala.js core (aka `java.lang.Object` in dependency tracking) depends on it.

**Comment by gzm0** [comment]
OK, I think I understand the problem better.

If we have a `ClassOf` tree in the body of say, `A#foo`. The analyzer marks that method to instantiate `java.lang.Class`. However, the `ClassOf` tree does not translate to a direct instantiation. In fact, it simply calls `getClass` on the class data. And the class data class is defined in the CoreJSLib, which is in a different module than `A`.

**Comment by gzm0** [comment]
Only known workaround is to disable the optimizer:

```
scalaJSLinkerConfig ~= (_.withOptimizer(false))
```

**Comment by gzm0** [comment]
(if we had implemented #1961, preventing `getClass` from being inlined would likely also fix the issue)
